uPortal-utils/uPortal-utils-core/src/main/java/org/apereo/portal/utils/personalize/PersonalizerImpl.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Licensed to Apereo under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright ownership. Apereo
+ * licenses this file to you under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the License at the
+ * following location:
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apereo.portal.utils.personalize;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpSession;
+import lombok.extern.slf4j.Slf4j;
+import org.apereo.portal.security.IPerson;
+import org.springframework.beans.factory.annotation.Value;
+
+/**
+ * Class that helps personalize text by replacing tokens with user attributes.
+ *
+ * <p>TODO - enable ehcache for mapped user attributes
```

Comment:
is this todo intended for this PR or a future one?

Reply:
interim direction.  I ended up using the http session object.  I'll remove the comment.